src/osp/Universe.h

```diff
@@ -173,6 +183,14 @@ TYPESAT_T& Universe::type_register(ARGS_T&& ... args)
     return toReturn;
 }
 
+template<typename SATTYPE_T>
+SATTYPE_T& Universe::sat_type_find()
+{
+    MapSatType::iterator it = sat_type_find(SATTYPE_T::smc_name);
+    assert(it != m_satTypes.end());
+    return static_cast<SATTYPE_T&>(*it->second);
+}
+
```

Comment:
```
template<typename TRAJECTORY_T, typename ... ARGS_T>
TRAJECTORY_T& Universe::trajectory_create(ARGS_T&& ... args)
{
    std::unique_ptr<TRAJECTORY_T> newTraj
            = std::make_unique<TRAJECTORY_T>(std::forward<ARGS_T>(args)...);
    TRAJECTORY_T& toReturn = *newTraj;
    m_trajectories.push_back(std::move(newTraj)); // this invalidates newTraj
    return toReturn;
}
```

I feel like this would simpler 

```
template<typename TRAJECTORY_T, typename ... ARGS_T>
TRAJECTORY_T& Universe::trajectory_create(ARGS_T&& ... args)
{
    return *( m_trajectories.emplace_back(std::make_unique<TRAJECTORY_T>(std::forward<ARGS_T>(args)...)) );
}
```

m_trajectories is an std::vector, so the emplace_back function returns a reference to the newly inserted item.